File: 底层班/13、第十三节课/dispatch_barrier/dispatch_barrier/ViewController.m
//
//  ViewController.m
//  dispatch_barrier
//
//  Created by chenghao on 2022/5/26.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        
    });
    
    // Do any additional setup after loading the view.
}


@end

File: Demo/HitTestDemo/hitTestDemo/hitTestDemo/ViewController.swift
//
//  ViewController.swift
//  hitTestDemo
//
//  Created by chenghao on 2022/9/2.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        let yellowView = YellowView.init(frame: CGRect.init(x: 10, y: 150, width: 400, height: 400))
        self.view.addSubview(yellowView)
    }
    
    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
        guard let touch = touches.first else { return }
        print("ViewController touch:\(touch.location(in: self.view))")
    }
    
    
    

}

